src/highlight/highlight.ts

```diff
@@ -22,15 +27,22 @@ export default class Highlight {
     this.agents = agents;
   }
 
-  postJoint(params: PostJointRequest) {
-    return this.asyncLock.acquire('postJoint', () => this._postJoint(params));
+  postMessage(request: PostMessageRequest) {
+    return this.asyncLock.acquire('postMessage', () =>
```

Comment:
Signature verification should not be part of the `asyncLock` function (`_postMessage()`). This can be done in parallel. It's much better specially for verifying contract sig which require external call, we don't want this to slowdown the process that must be done serially.